code/game/machinery/vending/marine_vending.dm

```diff
@@ -232,7 +238,62 @@
 	. = ..()
 	cargo_ammo_vendors.Remove(src)
 
+/obj/machinery/vending/lasgun
+	name = "ColMarTech Lasgun Field Charger"
+	desc = "An automated power cell dispenser and charger. Used to recharge energy weapon power cells, including in the field. Has an internal battery that charges off the power grid."
+	icon_state = "lascharger"
+	icon_vend = "lascharger-dispensing"
+	icon_deny = "lascharger-denied"
+	req_access = null
+	req_access_txt = "0"
+	req_one_access = null
+	req_one_access_txt = "9;2;21"
+	wrenchable = TRUE
+	drag_delay = FALSE
+	anchored = FALSE
+	idle_power_usage = 1
+	vend_power_usage = 50
+	machine_current_charge = 20000 //integrated battery for recharging energy weapons. Normally 10000.
+	machine_max_charge = 20000
+
+	product_ads = "Lasgun running low? Recharge here!;Need a charge?;Power up!;Electrifying!;Empower yourself!"
+	products = list(
+					/obj/item/ammo_magazine/lasgun/M43 = 10,
+					/obj/item/ammo_magazine/lasgun/M43/highcap = 2,
+					)
+
+	contraband =   list()
+
+	premium = list()
 
+	prices = list()
+
+/obj/machinery/vending/lasgun/New()
+	..()
+	update_icon()
+
+/obj/machinery/vending/lasgun/update_icon()
+	if(machine_current_charge <= 0)
```

Comment:
Use `switch(machine_current_charge*100/machine_max_charge)`

Reply:
Add a `if(machine_max_charge)` first, though, to avoid a possible runtime error if you try to divide by zero.